tests/timeseries/test_quantities.py

```diff
@@ -641,6 +641,247 @@ def test_invalid_multiplications() -> None:
             energy *= quantity  # type: ignore
 
 
+@pytest.mark.parametrize("quantity_ctor", _QUANTITY_CTORS + [Quantity])
+# Use a small amount to avoid long running tests, we have too many combinations
+@hypothesis.settings(max_examples=10)
+@hypothesis.given(
+    quantity_value=st.floats(
+        allow_infinity=False,
+        allow_nan=False,
+        allow_subnormal=False,
+        # We need to set this because otherwise constructors with big exponents will
+        # cause the value to be too big for the float type, and the test will fail.
+        max_value=1e298,
+        min_value=-1e298,
+    ),
+    percent=st.floats(allow_infinity=False, allow_nan=False, allow_subnormal=False),
+)
+def test_quantity_divided_by_precentage(
+    quantity_ctor: type[Quantity], quantity_value: float, percent: float
+) -> None:
+    """Test the division of all quantities by a percentage."""
+    hypothesis.assume(percent != 0.0)
+    percentage = Percentage.from_percent(percent)
+    quantity = quantity_ctor(quantity_value)
+    expected_value = quantity.base_value / (percent / 100.0)
+    print(f"{quantity=}, {percentage=}, {expected_value=}")
+
+    product = quantity / percentage
+    print(f"{product=}")
+    assert product.base_value == expected_value
+
+    quantity /= percentage
+    print(f"*{quantity=}")
+    assert quantity.base_value == expected_value
+
+
+@pytest.mark.parametrize("quantity_ctor", _QUANTITY_CTORS + [Quantity])
+# Use a small amount to avoid long running tests, we have too many combinations
+@hypothesis.settings(max_examples=10)
+@hypothesis.given(
+    quantity_value=st.floats(
+        allow_infinity=False,
+        allow_nan=False,
+        allow_subnormal=False,
+        # We need to set this because otherwise constructors with big exponents will
+        # cause the value to be too big for the float type, and the test will fail.
+        max_value=1e298,
+        min_value=-1e298,
+    ),
+    scalar=st.floats(allow_infinity=False, allow_nan=False, allow_subnormal=False),
+)
+def test_quantity_divided_by_float(
+    quantity_ctor: type[Quantity], quantity_value: float, scalar: float
+) -> None:
+    """Test the division of all quantities by a float."""
+    hypothesis.assume(scalar != 0.0)
+    quantity = quantity_ctor(quantity_value)
+    expected_value = quantity.base_value / scalar
+    print(f"{quantity=}, {expected_value=}")
+
+    product = quantity / scalar
+    print(f"{product=}")
+    assert product.base_value == expected_value
+
+    quantity /= scalar
+    print(f"*{quantity=}")
+    assert quantity.base_value == expected_value
```

Comment:
looks like we're missing tests for "valid" division by same quantity?

Reply:
Right! Good catch, I will add it.

Reply:
Oh, crap, this actually uncovered a (design) bug, good catch: `percentage / percentage` now have 2 possible types, `float` for `Self` and `Self` for `Percentage` :unamused: 

So there is no way to override the implementation in `Quantity` in a compatible way for `Percentage`. I guess `Percentage` being so special maybe shouldn't inherit from `Quantity`, but not sure. This is definitely out of scope for this PR, so I guess I will remove `Percentage` as a possible type for division.

Besides being an obscure use case, users can still do `quantity /= percentage.as_fraction()` if they really wish to divide by a percentage...